contrib/inventory/foreman.py

```diff
@@ -198,6 +198,25 @@ def _get_json(self, url, ignore_errors=None, params=None):
                 break
         return results
 
+    def _use_inventory_report(self):
+         url = "%s/api/v2/status" % self.foreman_url
+         result = self._get_json(url)
+         foreman_version = LooseVersion(result.get('version')) >= LooseVersion('1.24.0')
+         return foreman_version
+
+    def _post_request(self, url=None, params=None):
+        url = "%s/ansible/api/v2/ansible_inventories/schedule_report" % self.foreman_url
+        session = self._get_session()
+        params = {'input_values': {'required': ['all']}}
+        ret = session.post(url, json=params)
+        url = "{0}/api/v2/report_templates/153-Ansible%20Inventory/report_data/{1}".format(self.foreman_url, ret.json().get('job_id'))
+        while True:
+            ret = session.get(url)
```

Comment:
Probably cleaner to use a different variable than before to make sure you're not using the incorrect response.